### ISSUE TITLE
AMP Consent management for CCPA does not pass default-out-of-CA state

### ISSUE THREAD

**Issue by AramZS** [ampproject/amphtml#26081]
## What's the issue?

Glad to see an interface created for CCPA via #24910 however it is not entirely working in line with the expected behavior. Vendors expect to see the us_privacy string regardless of if users are applicable for CCPA coverage, because they need to know that the proper check has happened. As such, the ACTION_TYPE.DISMISS behavior at https://github.com/ampproject/amphtml/blob/e014a38fcacc5b94f58d461a7fa271ee3a2dc220/extensions/amp-consent/0.1/amp-consent.js#L413 is not compliant with the IAB specification. Even in a situation where the user cannot legally opt-out, they would still have a status (specifically `1---`) and that status needs to be passed to advertisers.

Additionally, documentation is highly unclear if `accept` is intended to reflect the state: "The User has NOT Opted Out" (1YNY) or the state "The User has Opted Out" (1YYY). Could you please make this clearer in either inline comments or documentation?

## How do we reproduce the issue?

Please provide a public URL and ideally a reduced test case (e.g. on jsbin.com) that exhibits only your issue and nothing else. Provide step-by-step instructions for reproducing the issue:

1. Go to any AMP that calls the consent API. 
2. Get a response that includes `"consentStateValue":"unknown"`

## What browsers are affected?

All.

## Which AMP version is affected?

Latest?

**Comment by AramZS** [comment]
Additionally, consent information should be sent even if it is 'the user is not valid for a consent check' (ex: `1---`) which means sending no signal on `consentRequired==false` is not expected behavior either.

**Comment by AramZS** [comment]
Finally, the documentation here: "publisher instructs AMP to remember the accept decision to the consent, unblocks components waiting for the consent, and hides the prompt UI." is unclear if `decision to consent` reflects the user activating an opt-out preference, or *not* opting out and needs clarity at https://github.com/ampproject/amphtml/blob/master/extensions/amp-consent/amp-consent.md

**Comment by zhouyx** [comment]
@jeffjose

**Comment by zhouyx** [comment]
>Additionally, consent information should be sent even if it is 'the user is not valid for a consent check' (ex: 1---) which means sending no signal on consentRequired==false is not expected behavior either.

One question about the `consentRequired==false` state. If in the future we are providing state level geo detection. Would such a string still be required if no request to the `checkConsentHref` is required? Aka do you need this string passed to vendor if the user is not from EEA or US today? Thanks

**Comment by jeffjose** [comment]
If you set `consentStateValue` to `accept` for residents outside of CA and pass `1---` string, wouldnt that work?

Since CCPA only applies to CA, the above treats residents outside of CA as having `accept`ed this consent.

**Comment by lannka** [comment]
@AramZS according to the design that we all agreed on, consent string will be collected by your server and passed back to AMP in the checkConsentHref response. Would that not work?

**Comment by AramZS** [comment]
@jeffjose @lannka Our issue is that `consentStateValue=="accept"` is unclear, we can set that to `accept` and pass `1---`. Does that signal the user has "*not* opted out"? The language was unclear and that is my concern. I understand that we're expanding an existing component, but it seemed odd to indicate that a user had consented when it was just that they weren't eligible to opt out. 

I understand this is not how it currently works, but it seemed to me that the string could be set and passed regardless of `consentStateValue`, since `undefined` is technically correct as a description of a user's consent state. 

And yes, we still need the string passed to the vendor if the user is not from EEA or US, it is used by downstream vendors to understand that we have checked to see if CCPAA applied and determined that it did not, rather than failing to check if the user was eligible. Downstream systems need to know that we have fulfilled our responsibility in that regard and this signal is the way to indicate that. 

I'm unclear about 

> Would such a string still be required if no request to the checkConsentHref is required?

I assume that refers to the same situation I addressed above (the user has been checked for eligibility to a consent law and determined to not be within its legal purview)? If so, yes, if the geo does not require that the user have a consent state, we still need `1---` which is set explicitly to show that we have made that determination. We can send you that string in the API.
